src/prefect/agent/kubernetes/agent.py

```diff
@@ -10,9 +12,51 @@
 from prefect import config
 from prefect.agent import Agent
 from prefect.engine.state import Failed
-from prefect.utilities.agent import get_flow_image, get_flow_run_command
+from prefect.serialization.run_config import RunConfigSchema
+from prefect.utilities.agent import (
+    get_flow_image,
+    get_flow_run_command,
+    get_flow_image_if_docker_storage,
+)
 from prefect.utilities.graphql import GraphQLResult
 
+DEFAULT_JOB_TEMPLATE_PATH = os.path.join(os.path.dirname(__file__), "job_template.yaml")
+
+
+def _get_or_create(d: dict, key: str, val: Any = None) -> Any:
+    if val is None:
+        val = {}
+    path = key.split(".")
+    for k in path[:-1]:
+        d = d.setdefault(k, {})
+    return d.setdefault(path[-1], val)
+
+
+def read_bytes_from_path(path: str) -> bytes:
```

Comment:
Is this the sort of logic that using `fsspec` would take care of (ignoring the authentication patterns)?

Reply:
The logic here is so short I don't think it's too bad, and since prefect has its own credentials convention for s3/gcs, relying on fsspec wouldn't get us much here. Note that the interface doesn't prevent switching to using fsspec transparently later, but for now I don't see the need.